client/src/app/add/add-note.component.ts

```diff
@@ -0,0 +1,62 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormControl, FormGroup, Validators } from '@angular/forms';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { Router } from '@angular/router';
+import { Note } from '../note';
+import { NotesService } from '../notes.service';
+
+@Component({
+  selector: 'app-add-note',
+  templateUrl: './add-note.component.html',
+  styleUrls: ['./add-note.component.scss']
+})
+export class AddNoteComponent implements OnInit {
+
+  addNoteForm: FormGroup;
+
+  note: Note;
+
+  constructor(private fb: FormBuilder, private noteService: NotesService, private snackBar: MatSnackBar, private router: Router) {
+  }
+
+  // not sure if this name is magical and making it be found or if I'm missing something,
+  // but this is where the red text that shows up (when there is invalid input) comes from
+  add_note_validation_messages = {
+    body: [
+      {type: 'required', message: 'Body is required'},
+      {type: 'minlength', message: 'Body must be at least 2 characters long'},
+      {type: 'maxlength', message: 'Body cannot be more than 50 characters long'}
+    ]
+  };
+
+  createForms() {
+
+    // add note form validations
+    this.addNoteForm = this.fb.group({
+      body: new FormControl('', Validators.compose([
+        Validators.required,
+        Validators.minLength(2),
+        Validators.maxLength(250),
```

Comment:
And then down here you say the maximum length is `250`, not `50`.

Reply:
Can we just bump it up to an even thousand? That's about two paragraphs.

Reply:
(You'd need to change it in the e2e tests as well.)

Reply:
(And possibly the Karma tests)

Reply:
Update: Audrey recommends `280` characters because that's the length of a tweet 🙃

Reply:
(And the server)

Reply:
You know, that might be a lot of work. I'm fine with leaving it at 250.

Reply:
I'll increase it to 300